src/main/java/de/zalando/aruha/nakadi/repository/kafka/KafkaRepository.java

```diff
@@ -99,6 +94,30 @@ public void createTopic(final String topic, final int partitionsNum, final int r
         }
     }
 
+    public boolean topicExists(final String topic) throws NakadiException {
+        return listTopics()
+                .stream()
+                .map(Topic::getName)
+                .anyMatch(t -> t.equals(topic));
+    }
+
+    public boolean areCursorsValid(final String topic, final List<Cursor> cursors) {
+        final List<TopicPartition> partitions = listPartitions(topic);
+        return cursors
+                .stream()
+                .allMatch(cursor -> partitions
+                        .stream()
+                        .filter(tp -> tp.getPartitionId().equals(cursor.getPartition()))
+                        .findFirst()
+                        .map(pInfo -> {
+                            final long newestOffset = Long.parseLong(pInfo.getNewestAvailableOffset());
+                            final long oldestOffset = Long.parseLong(pInfo.getOldestAvailableOffset());
+                            final long offset = Long.parseLong(cursor.getOffset());
```

Comment:
This can potentially throw a `NumberFormatException` if the user specifies a non-number String in the header. A `NumberFormatException` here will result into a `500 Internal Server Error`. Instead it should be sth `4xx` (according to the api).